examples/arduino_keyboard.rs

```diff
@@ -118,19 +122,12 @@ struct UsbContext {
 impl UsbContext {
     fn poll(&mut self) {
         if self.trigger.is_low() {
-            if let Some(report) = PAYLOAD
-                .get(self.current_index)
-                .copied()
-                .and_then(ascii_to_report)
-            {
-                if self.hid_class.push_input(&report).is_ok() {
-                    self.current_index += 1;
-                }
-            } else {
-                self.hid_class.push_input(&BLANK_REPORT).ok();
+            let report = ascii_to_report(PAYLOAD[self.current_index]).unwrap_or(BLANK_REPORT);
+
+            if self.hid_class.push_input(&report).is_ok() {
```

Comment:
> Just to make sure, are you checking the result of push_input()?

Yes, I don't advance the payload index unless the key report is successful.

Below, I delay for `5ms` unconditionally as a crude/simple debouncer, then send a blank report. I don't check the result of `push_input` (discard the result with a `.ok()` call).

If the following poll is successful, I use what was here previously, i.e. check the `raw_output` result to conditionally set the `indicator`.